src/Smalot/PdfParser/Pages.php

```diff
@@ -57,17 +62,65 @@ public function getPages(bool $deep = false): array
             return $kidsElement->getContent();
         }
 
+        // Prepare to apply the Pages' object's fonts to each page
+        $fonts = $this->getFonts();
+
         $kids = $kidsElement->getContent();
         $pages = [];
 
         foreach ($kids as $kid) {
             if ($kid instanceof self) {
                 $pages = array_merge($pages, $kid->getPages(true));
             } elseif ($kid instanceof Page) {
+                if (!empty($this->fonts)) {
+                    $kid->setFonts($fonts);
+                }
                 $pages[] = $kid;
             }
         }
 
         return $pages;
     }
+
+    /**
+     * @return Font[]
+     */
+    public function getFonts()
+    {
+        if (null !== $this->fonts) {
+            return $this->fonts;
+        }
+
+        $resources = $this->get('Resources');
+
+        if (method_exists($resources, 'has') && $resources->has('Font')) {
+            if ($resources->get('Font') instanceof Element\ElementMissing) {
+                return [];
+            }
+
+            if ($resources->get('Font') instanceof Header) {
+                $fonts = $resources->get('Font')->getElements();
+            } else {
+                $fonts = $resources->get('Font')->getHeader()->getElements();
+            }
+
+            $table = [];
+
+            foreach ($fonts as $id => $font) {
+                if ($font instanceof Font) {
+                    $table[$id] = $font;
+
+                    // Store too on cleaned id value (only numeric)
+                    $id = preg_replace('/[^0-9\.\-_]/', '', $id);
+                    if ('' != $id) {
+                        $table[$id] = $font;
+                    }
+                }
+            }
+
+            return $this->fonts = $table;
+        }
```

Comment:
Please give me a short summary of this part. A few comments in the code would also help.

Why is it public if you only use it internally?

Reply:
This is roughly a copy of the same method on the Page class.  Perhaps that function should be factored out into a utility class.